conn2res/tasks.py

```diff
@@ -160,16 +161,23 @@ def fetch_data(self, n_trials=None, **kwargs):
             env.new_trial()
             ob, gt = env.ob, env.gt
 
-            # store inputs
+            # reshape data if needed
             if ob.ndim == 1:
-                x.append(ob[:, np.newaxis])
-            else:
-                x.append(ob)
-
+                ob = ob[:, np.newaxis]
             if gt.ndim == 1:
-                y.append(gt[:, np.newaxis])
-            else:
-                y.append(gt)
+                gt = gt[:, np.newaxis]
+
+            # scale input data
+            if input_gain is not None:
+                ob *= input_gain
```

Comment:
I moved input_gain from the simulate method of reservoirs to the fetch_data method of tasks as it is actually the signal we want to adjust instead of the input weights of the reservoir. For instance, this allows to scale the input data whilst keeping the bias (see just below) unchanged